pytomorrowio/pytomorrowio.py

```diff
@@ -276,22 +286,26 @@ async def realtime_and_all_forecasts(
         `forecast_or_nowcast_fields` will be used to get nowcast, hourly, and daily
         data.
         """
-        ret_data = {}
-        data = await self._call_api(
-            {
-                "timesteps": ["current"],
-                "fields": realtime_fields,
-            }
-        )
-        if (
-            "data" in data
-            and "timelines" in data["data"]
-            and "intervals" in data["data"]["timelines"][0]
-            and "values" in data["data"]["timelines"][0]["intervals"][0]
-        ):
-            ret_data[CURRENT] = data["data"]["timelines"][0]["intervals"][0]["values"]
-
-        forecasts = ret_data.setdefault(FORECASTS, {})
+        self._num_api_requests = 0
+        ret_data: dict[str, dict[str, Any]] = {CURRENT: {}, FORECASTS: {}}
+        for i in range(0, len(realtime_fields), MAX_FIELDS):
```

Comment:
the point here is to support more than `MAX_FIELDS` by iterating through `MAX_FIELDS` at a time and collating the data. I have a dependent PR to add a number of new sensors that will cause us to exceed the limit

Reply:
Gotcha

Reply:
ah I see the concern - you are saying that `available_fields` returns less than 55 in all cases. I think I'm going to get rid of `available_fields`. When I initially started working on this, I wanted to encode some intelligence into the library about fields, but it's tedious and easier to handle upstream.

Reply:
hmm I forgot about the measurements piece, so may need to hold off on this for now

Reply:
It's a tricky problem. We definitely want to minimize the number of API calls for free API users (like me). But the difference in data required for hourly and daily forecasts pretty big imo.

Reply:
can you give me some examples? It's been so long since I looked at all the different forecast types that it's no longer top of mind so I would have to relook it up

Reply:
all of the attributes that `homeassistant` supports natively aren't all that different between the different forecast types. I guess down the road we may want to add more forecast attributes, but seems like a problem to tackle when we get there. For now, with the measurements we are using, we can use the same field list for both hourly and daily I think, but let me know if you think I am missing something